### ISSUE TITLE
[Xamarin iOS] Marker rendered completely white without color

### ISSUE THREAD

**Issue by BamButz** [CartoDB/mobile-sdk#442]
## Description

When adding markers or points with custom bitmap as described in [vector objects on map](https://carto.com/developers/mobile-sdk/guides/vector-objects-on-map/) without specifying a color, the object gets rendered completely white.

Using the same logic on android or UWP renders the bitmap correct.

#### iOS
<img src="https://user-images.githubusercontent.com/7022144/127318940-b17a14f2-a843-45f9-a1ae-444f543fa7b1.png" height="512">

#### Android
<img src="https://user-images.githubusercontent.com/7022144/127320390-036b77b9-836c-4dc0-a9c6-4b514a4b7a99.png" height="512">

## Code Sample
`marker.Bitmap` is filled by `BitmapUtils.CreateBitmapFromUIImage` on iOS and `BitmapUtils.CreateBitmapFromAndroidBitmap` on Android

``` csharp
PointStyleBuilder pointStyleBuilder = new PointStyleBuilder();
pointStyleBuilder.Size = 32;
pointStyleBuilder.Bitmap = marker.Bitmap;

PointStyle pointStyle = pointStyleBuilder.BuildStyle();

MapPos pos = proj.FromLatLong(marker.Position.Latitude, marker.Position.Longitude);
Point point = new Point(pos, pointStyle);
_localVectorDataSource.Add(point);
```

**Comment by mtehver** [comment]
My guess is that your project has the following option turned on: 'Optimize PNG images' and your marker image is a PNG file. This option should be turned off, as when turned on, PNG files are transcoded into proprietary format that SDK is unable to read. There should be a warning about this in the SDK log, also.

**Comment by BamButz** [comment]
You're talking about the optimization process during compilation?

If yes, I have to disappoint you, we are painting the images via SkiaSharp and then converting them from SKBitmap to UIImage/Android.Bitmap and then via BitmapUtils to Carto.Graphics.Bitmap.
I mean, the proprietary format may also be used during the conversion, but wouldn't it also be used then without compile time optimization?

**Comment by mtehver** [comment]
Correct, I originally assumed you are using PNG assets for your markers.

Could you share the code that constructs `marker.bitmap`? It is possible that some variants of `UIImage` encodings are not properly converted into internal bitmap data, but we need to have a reproducible testcase to investigate this.

A workaround would be to bypass `UIImage` conversion from `SKImage` completely, by doing something like this:

```
var skImage = ...
var bitmapData = new BinaryData(skImage.Encode(SKEncodedImageFormat.Png, 100).ToArray());
pointStyleBuilder.Bitmap = Carto.Graphics.Bitmap.CreateFromCompressed(bitmapData);
```

**Comment by mtehver** [comment]
This should be fixed now, fix is in 4.4.1RC1.

**Comment by mtehver** [comment]
4.4.1 is released now, thus closing.
